formats/json/jsMain/src/kotlinx/serialization/json/internal/DynamicEncoders.kt

```diff
@@ -58,6 +58,8 @@ private class DynamicObjectEncoder(
     private lateinit var currentDescriptor: SerialDescriptor
     private var currentElementIsMapKey = false
 
+    private var objectPolymorphism = false
```

Comment:
I'd suggest to rename this flag to `writePolymorphic` as it is named in other implementation because it is doing basically the same thing.

Reply:
OK, but I think name writePolymorphic is incorrect - it is only true when the discriminator attribute is used